JustLog/Classes/LogstashDestination.swift

```diff
@@ -80,11 +81,12 @@ public class LogstashDestination: BaseDestination  {
         return nil
     }
 
-    public func forceSend(_ completionHandler: @escaping (_ error: Error?) -> Void  = {_ in }) {
-
-        self.completionHandler = completionHandler
-
-        writeLogs()
+    public func forceSend(_ completionHandler: @escaping (_ error: Error?) -> Void = {_ in }) {
+        logDispatchQueue.addOperation { [weak self] in
+            guard let self = self else { return }
+            self.completionHandler = completionHandler
+            self.writeLogs()
```

Comment:
`writeLogs` also adds an operation to the queue, is it ok to do this inside of an `addOperation` block?

Reply:
I had to think about this a bit too. But I couldn't find anything against it and Apple's documentation says that's safe to use an operation queue from different thread [here](It is safe to use a single OperationQueue object from multiple threads without creating additional locks to synchronize access to that object.). 
I checked the code on an example code that crashes without this fix and it does not crash after I implemented this change anymore.
It looks a bit weird, but at the end is the same as doing a `DispatchQueue.main.async` inside another `DispatchQueue.main.async`, since they are just scheduling a block in the future they don't even need to be reentrant because one is running while scheduling the other.

Reply:
I don't think it would make much difference if we call `writeLogs` from within the block or outside since we have `maxConcurrentOperationCount = 1` for the queue. Makes me think the implementation of `forceSend` could be as follow though:

```
self.completionHandler = completionHandler
writeLogs()
```

Reply:
Shame on me! I thought too much and re-proposed the original code! 🙈

Reply:
Shame on me! I thought too much and proposed the original code! 🙈